### PR TITLE
tests: no #pragma for gcc < 4.6

### PR DESCRIPTION
**Pull request by gperciva**

### CODE REVIEW

**Comment by gperciva** [comment]
In particular, OpenBSD 6.1's cc is gcc 4.2.1 (?!?), which cannot handle `#pragma` inside a function.

**Comment by gperciva** [comment]
Better solution, including fixing the build.

File: tests/parsenum/main.c [hunk 1]
```diff
@@ -102,7 +102,7 @@ main(int argc, char * argv[])
 
 #if __clang__
 	_Pragma("clang diagnostic ignored \"-Wtautological-constant-out-of-range-compare\"");
-#elif __GNUC__
+#elif __GNUC__ >= 4 && __GNUC_MINOR__ >= 6
```
Comment (cperciva) [inline tests/parsenum/main.c]: So, err, we don't want to include this code for GCC 5.1?

**Comment by gperciva** [comment]
Yikes, how embarrassing!  Updated PR.

**Pull request merged**